lib/shopify_app/controller_concerns/csrf_protection.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+module ShopifyApp
+  module CsrfProtection
+    extend ActiveSupport::Concern
```

Comment:
Since you're using `ActiveSupport::Concern`, you can just `include LoginProtection` here. You don't need to do the ancestors check.

See https://api.rubyonrails.org/classes/ActiveSupport/Concern.html

Reply:
I'm not sure that's a good idea. `CsrfProtection` should ideally have nothing to do with `LoginProtection`. They serve different purposes. If you want, I can have `CsrfProtection` access the `jwt_shopify_domain` directly from the request environment.

Reply:
I think the middleware exposes that interface right now, so it doesn't bother me if you access the `env` directly.

My feedback is that under the current approach you have a dependency on `LoginProtection`, and it's better to use the tools that `ActiveSupport::Concern` gives you to enforce that dependency than to build your own thing.

Reply:
Would it make sense to do something more duck-typing-esque? Like, put some defensive code around the reference to `jwt_shopify_domain` to raise a meaningful error if it's not present?

Reply:
I don't think so, personally. IMO the interface is the keys added to the `env`. We can either access that interface directly or we can pull in a method we know accesses the interface correctly.

The `jwt_shopify_domain` method is not the interface and testing that a `jwt_shopify_domain` method of unknown origin is present is the wrong approach IMO.

Reply:
The thing is we don't know if the app is using JWT or cookie based auth (Especially due to the fallbacks where it can use either). So we default to the standard Rails CSRF error.